daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1067,6 +1067,9 @@ private[lf] object SBuiltin {
             .foreach(nid =>
               throw SErrorDamlException(IE.ContractNotActive(coid, cached.templateId, nid))
             )
+          if (!onLedger.ptx.isActive(coid)) {
```

Comment:
shouldn’t we be able to drop the `onLedger.ptx.consumedBy` above? This looks like we’re now doing two activeness check.

Reply:
You get a more precise error message from the original check.

Reply:
Can we make isActive return an `Either[NodeId, ()]` or something like that if it fails due to `consumedBy` so we get the more precise error in that case? I don’t think we’re doing us any favor by keeping two sets of check.

Reply:
ok